crates/sui/src/client_commands.rs

```diff
@@ -415,15 +417,31 @@ impl SuiClientCommands {
                 let sender = context.try_get_object_owner(&gas).await?;
                 let sender = sender.unwrap_or(context.active_address()?);
 
-                let compiled_modules = build_move_package(
+                let compiled_package = build_move_package(
                     &package_path,
                     BuildConfig {
                         config: build_config,
                         run_bytecode_verifier: true,
                         print_diags_to_stderr: true,
                     },
-                )?
-                .get_package_bytes();
+                )?;
+
+                let compiled_modules = compiled_package.get_package_bytes();
+
+                // verify that all dependency packages have the correct on-chain bytecode
+                #[cfg(not(msim))]
```

Comment:
Noting our conversation offline about this here -- we'll need to find a way to make this work with `msim` because this is the configuration that the majority of `sui`'s tests are built and run with, so disabling this functionality in that setting is going to really hurt test coverage on the verifier.

@mystenmark, the issue seems to be that `msim` uses `sui_sdk::embedded_gateway::SuiClient` instead of `sui_sdk::SuiClient` for the client, and that is causing a build issue in this PR -- is that just part of how `simtest` works, or is this an artefact of `simtest` needing to be updated after the Gateway deprecation?